### PR TITLE
 initial support for haproxy 2.0

### PR DESCRIPTION
**Pull request by JohnRoesler**
### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by Xorima** [comment]
I'm not codeowners :D

File: resources/install.rb [hunk 1]
```diff
@@ -15,16 +15,16 @@
...
+property :source_checksum, [String, nil], default: 'fe0a0d69e1091066a91b8d39199c19af8748e0e872961c6fc43c91ec7a28ff20'
 property :source_target_cpu, [String, nil], default: lazy { node['kernel']['machine'] }
 property :source_target_arch, [String, nil], default: nil
 property :source_target_os, String, default: lazy {
   if node['kernel']['release'].split('.')[0..1].join('.').to_f > 2.6
-    @target_os = 'linux2628'
+    @target_os = source_version.chars.first == '1' ? 'linux2628' : 'linux-glibc'
   elsif (node['kernel']['release'].split('.')[0..1].join('.').to_f > 2.6) && (node['kernel']['release'].split('.')[2].split('-').first.to_i > 28)
-    @target_os = 'linux2628'
+    @target_os = source_version.chars.first == '1' ? 'linux2628' : 'linux-glibc'
```
Comment (Xorima) [inline resources/install.rb]: This should be a library?

File: test/integration/source-default/source_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+describe directory '/etc/haproxy' do
+  it { should exist }
+end
+
+describe file '/etc/haproxy/haproxy.cfg' do
+  its(:mode) { should cmp '0644' }
+end
+
+if os.debian? && os.release.start_with?('14')
```
Comment (Xorima) [inline test/integration/source-default/source_spec.rb]: why is this different from others? why would it not be enabled on 14?

File: resources/install.rb (hunk 1 above)
Comment (JohnRoesler) [reply resources/install.rb]: yeah 👍 not a codeowner, but maybe you should be ... 😉

File: test/integration/source-default/source_spec.rb (hunk 2 above)
Comment (JohnRoesler) [reply test/integration/source-default/source_spec.rb]: relic of the past - we don't even support 14, dropping this

File: resources/install.rb (hunk 1 above)
Comment (Xorima) [reply resources/install.rb]: even if we only use this once it is a lot of logic for a default in the resource, where as in a library it seems more sane? I could be wrong though

File: libraries/helpers.rb [hunk 3]
```diff
@@ -33,9 +33,7 @@ def ius_package
       end
 
       def target_os(source_version)
-        if node['kernel']['release'].split('.')[0..1].join('.').to_f > 2.6
-          return source_version.chars.first == '1' ? 'linux2628' : 'linux-glibc'
-        elsif (node['kernel']['release'].split('.')[0..1].join('.').to_f > 2.6) && (node['kernel']['release'].split('.')[2].split('-').first.to_i > 28)
+        if (node['kernel']['release'].split('.')[0..1].join('.').to_f > 2.6) && (node['kernel']['release'].split('.')[2].split('-').first.to_i > 28)
```
Comment (Xorima) [inline libraries/helpers.rb]: what if equal to 28 exactly?

**Pull request merged**